cunumeric/module.py

```diff
@@ -6227,3 +6227,728 @@ def bincount(
+    interpolated_inverted_cdf:
+        method 4 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 1
+    hazen:
+        method 5 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1/2
+        * beta = 1/2
+    weibull:
+        method 6 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 0
+    linear:
+        method 7 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1
+        * beta = 1
+    median_unbiased:
+        method 8 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is unknown (see reference).
+        This method gives continuous results using:
+        * alpha = 1/3
+        * beta = 1/3
+    normal_unbiased:
+        method 9 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is known to be normal.
+        This method gives continuous results using:
+        * alpha = 3/8
+        * beta = 3/8
+    lower:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` as the interpolation point.
+    higher:
+        NumPy method kept for backwards compatibility.
+        Takes ``j`` as the interpolation point.
+    nearest:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` or ``j``, whichever is nearest.
+    midpoint:
+        NumPy method kept for backwards compatibility.
+        Uses ``(i + j) / 2``.
+    References
+    ----------
+    .. [1] R. J. Hyndman and Y. Fan,
+       "Sample quantiles in statistical packages,"
+       The American Statistician, 50(4), pp. 361-365, 1996
+    """
+
+    from .logic import isscalar
+
+    dict_methods = {
+        "inverted_cdf": inverted_cdf,
+        "averaged_inverted_cdf": averaged_inverted_cdf,
+        "closest_observation": closest_observation,
+        "interpolated_inverted_cdf": interpolated_inverted_cdf,
+        "hazen": hazen,
+        "weibull": weibull,
+        "linear": linear,
+        "median_unbiased": median_unbiased,
+        "normal_unbiased": normal_unbiased,
+        "lower": lower,
+        "higher": higher,
+        "midpoint": midpoint,
+        "nearest": nearest,
+    }
+
+    axes_set = []
+    original_shape = a.shape
+
+    if (axis is not None) and (not isscalar(axis)):
+        if len(axis) == 1:
+            real_axis = axis[0]
+            a_rr = a
+            axes_set = [real_axis]
+        else:
+            (real_axis, a_rr) = reshuffle_reshape(a, axis)
+            # What happens with multiple axes and overwrite_input = True ?
+            # It seems overwrite_input is reset to False;
+            overwrite_input = False
+            axes_set = axis
+    else:
+        real_axis = axis
+        a_rr = a
+        if axis is None:
+            axes_set = original_shape
```

Comment:
Shouldn't this be `list(range(a.ndim))`? `a.shape` is the extents of the dimensions, e.g. it would be `(5,5,5)` for a 5 x 5 x 5 array, but IIUC you want this to be the indices over which the quantile is taken, i.e. in this case it should be `(0,1,2)`, not `(5,5,5)`.

Reply:
`axes_set` is the set of axes along which quantiles are computed. 

Except for the case above (`axis is None`) it has nothing to do with the input array (shape, or anything else).

Now, for the `axis is None` case I'm passing the `original shape` of the array, bc/ those dimensions must be reset to 1 for `keepdims==True`. So, the reason I'm passing `original_shape` is because I want to replace it with as many 1's as there are elements in the `original_shape`. Which is what happens in the line:
```
remaining_shape = tuple(ones(len(axes_set), dtype=int))
```

Reply:
Which I've fixed, just now.

Reply:
I'm not talking about `original_shape` itself here, I see that `original_shape` is passed as a separate argument to `quantile_impl`, and that looks right to me.

I am talking about `axes_set`, which is a different argument to `quantile_impl`. I think setting `axes_set` to `original_shape` is wrong. When `original_shape` is `(42,43,44)` and `axis` is `None`, then `axes_set` should be `(0,1,2,)`, not `(42,43,44)`, if I'm understanding the purpose of this argument correctly.